### DISCUSSION TITLE
webERP 4.15.2 Deprecated: Methods with the same name as their class will not be constructors in a future version of PHP; PHPlot has a deprecated constructor in /weberp/includes/phplot/phplot.php on line 36

### DISCUSSION THREAD

**Discussion by pakricard** [timschofield/webERP#207] [Problems / Bugs] [answered]
Hi:

Testing SalesGraph.php, I saw this warning: Deprecated: Methods with the same name as their class will not be constructors in a future version of PHP; PHPlot has a deprecated constructor in /weberp/includes/phplot/phplot.php on line 36
webERP v 4.15.2 still uses PHPlot 5.5,  it needs to be updated to PHPlot 6.2

I tested my webERP installation with PHPPlot 6.2 (just needed to download the v6.2 from https://sourceforge.net/projects/phplot/, delete all current files and copy the new files in weberp/includes/phplot and it worked.

**Comment by timschofield** [comment, marked as answer]
Thanks Ricard, have updated the master branch with this.

**Comment by dalers** [comment]
@pakricard @timschofield related to commit 1ae8a562600066f9900753370697a762c411f82c  I researched PHPlot and agree it seems unmaintained since Aug 2023.

Fwiw, the README in the https://github.com/AJRepo/PHPlot repo reports reference documentation is on https://github.com/PHPlot/phplot, implying it's "now" the master project repo. Fwiw, it has one more commit than the AJRepo/PHPlot repo ("Fix soft link in phplotdocs").

There are two open PRs to the AJRepo/PHPlot repo. One was committed seperately to PHPlot/phplot (the additional commit), but the second PR ("phplot.php: type casts for PHP 8.2") could be worth duplicating in webERP (I presume you will know better than I).

I also checked the AJRepo/PHPlot and PHPlot/phplot forks, two have commits that could be worth duplicating in webERP:

- https://github.com/thepoorswiss/PHPlot - "Add basic support for subtitles"
- https://github.com/wp-statistics/PHPlot - two "minor fixes" (appear related to PHP8) and "PHP 8.2 deprecation warnings"

Unless we want to switch out PHPlot for an actively maintained project (which would take work, there are 516 occurences of "phplot" in 29 files), it seems likely we will be on our own maintaining PHPlot. Perhaps plan on a switch to something else for webERP v6, or sooner if any insurmountable issues are encountered?

**Comment by pakricard** [comment]
Hi @dalers :

I downloaded and installed the PHPlot from https://github.com/wp-statistics/PHPlot: it mentions that it is @version 8.0.0-rc1and  has 6 commits ahead of the https://github.com/AJRepo/PHPlot repository I found. Thanks!

But after installing it, I got a 
Fatal error: Uncaught Error: Class "PHPlot" not found in /weberp/SalesGraph.php:268 Stack trace: #0 {main} thrown in /weberp/SalesGraph.php on line 268,
so probably it is the right way to go (long term), but it implies a rewrite of SalesGraph.php out of my knowledge.

My two cents:
At this time, I think we have too many issues to fix (many from v4.15.2 that should be stable), and adding this one is a tiny advancement as it is only used in one script in standard webERP. 

I guess we can live with some deprecated warnings here (using the PHPlot 6.2) and once webERP v5 is mature and stable, we can start with this PHPlot upgrade or change to a new graph library.

**Comment by timschofield** [comment]
I think I agree that at this stage I can live with some deprecated warnings
at this stage with a view to later on moving to a different library.

Thanks
Tim

On Mon, 2 Dec 2024 at 02:27, pakricard ***@***.***> wrote:

> Hi @dalers <https://github.com/dalers> :
>
> I downloaded and installed the PHPlot from
> https://github.com/wp-statistics/PHPlot: it mentions that it is @Version
> <https://github.com/Version> 8.0.0-rc1and has 6 commits ahead of the
> https://github.com/AJRepo/PHPlot repository I found. Thanks!
>
> But after installing it, I got a
> Fatal error: Uncaught Error: Class "PHPlot" not found in
> /weberp/SalesGraph.php:268 Stack trace: #0 {main} thrown in
> /weberp/SalesGraph.php on line 268,
> so probably it is the right way to go (long term), but it implies a
> rewrite of SalesGraph.php out of my knowledge.
>
> My two cents:
> At this time, I think we have too many issues to fix (many from v4.15.2
> that should be stable), and adding this one is a tiny advancement as it is
> only used in one script in standard webERP.
>
> I guess we can live with some deprecated warnings here (using the PHPlot
> 6.2) and once webERP v5 is mature and stable, we can start with this PHPlot
> upgrade or change to a new graph library.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/207#discussioncomment-11431737>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LFV2N274TBBSUPT54L2DPAQVAVCNFSM6AAAAABRFRLC56VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNBTGE3TGNY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
www.weberp.org <http://www.weberpafrica.com>
@TimSchofield2
Blog: https://kwamoja.home.blog/
